core/src/main/scala/com/avast/clients/rabbitmq/DefaultRabbitMQClientFactory.scala

```diff
@@ -31,262 +23,87 @@ private[rabbitmq] object DefaultRabbitMQClientFactory extends LazyLogging {
-        cs: ContextShift[F]): DefaultRabbitMQConsumer[F] = {
-
-      val mergedConfig = providedConfig.withFallback(ConsumerDefaultConfig)
-
-      // merge consumer binding defaults
-      val updatedConfig = {
-        val updated =
-          mergedConfig.as[Seq[Config]]("bindings").map { bindConfig =>
-            bindConfig.withFallback(ConsumerBindingDefaultConfig).root()
-          }
-
-        import scala.collection.JavaConverters._
-
-        mergedConfig.withValue("bindings", ConfigValueFactory.fromIterable(updated.asJava))
-      }
-
-      val consumerConfig = updatedConfig.wrapped(configName).as[ConsumerConfig](configName)
-
-      create[F, A](consumerConfig, configName, channel, channelFactoryInfo, blocker, monitor, consumerListener, readAction)
-    }
-
     def create[F[_]: ConcurrentEffect, A: DeliveryConverter](
         consumerConfig: ConsumerConfig,
-        configName: String,
         channel: ServerChannel,
-        channelFactoryInfo: RabbitMQConnectionInfo,
+        connectionInfo: RabbitMQConnectionInfo,
         blocker: Blocker,
         monitor: Monitor,
         consumerListener: ConsumerListener,
         readAction: DeliveryReadAction[F, A])(implicit timer: Timer[F], cs: ContextShift[F]): DefaultRabbitMQConsumer[F] = {
 
-      prepareConsumer(consumerConfig, configName, readAction, channelFactoryInfo, channel, consumerListener, blocker, monitor)
+      prepareConsumer(consumerConfig, readAction, connectionInfo, channel, consumerListener, blocker, monitor)
     }
   }
 
   object PullConsumer {
 
-    def fromConfig[F[_]: ConcurrentEffect, A: DeliveryConverter](
-        providedConfig: Config,
-        configName: String,
-        channel: ServerChannel,
-        channelFactoryInfo: RabbitMQConnectionInfo,
-        blocker: Blocker,
-        monitor: Monitor)(implicit cs: ContextShift[F]): DefaultRabbitMQPullConsumer[F, A] = {
-
-      val mergedConfig = providedConfig.withFallback(PullConsumerDefaultConfig)
-
-      // merge consumer binding defaults
-      val updatedConfig = {
-        val updated =
-          mergedConfig.as[Seq[Config]]("bindings").map { bindConfig =>
-            bindConfig.withFallback(ConsumerBindingDefaultConfig).root()
-          }
-
-        import scala.collection.JavaConverters._
-
-        mergedConfig.withValue("bindings", ConfigValueFactory.fromIterable(updated.asJava))
-      }
-
-      val consumerConfig = updatedConfig.wrapped(configName).as[PullConsumerConfig](configName)
-
-      create[F, A](consumerConfig, configName, channel, channelFactoryInfo, blocker, monitor)
-    }
-
     def create[F[_]: ConcurrentEffect, A: DeliveryConverter](
         consumerConfig: PullConsumerConfig,
-        configName: String,
         channel: ServerChannel,
-        channelFactoryInfo: RabbitMQConnectionInfo,
+        connectionInfo: RabbitMQConnectionInfo,
         blocker: Blocker,
         monitor: Monitor)(implicit cs: ContextShift[F]): DefaultRabbitMQPullConsumer[F, A] = {
```

Comment:
should be
```scala
: DefaultRabbitMQPullConsumer[F, A] = {
```

Reply:
I disagree, this is private object which serves as factory for a specific implementation and as such it's allowed to return specific type.